vpn/src/main/java/com/duckduckgo/mobile/android/vpn/service/VpnUncaughtExceptionHandler.kt

```diff
@@ -53,6 +61,22 @@ class VpnUncaughtExceptionHandler(
             TrackerBlockingVpnService.restartVpnService(context, forceGc = true)
         }
     }
+
+    private fun recordExceptionAndAllowCrash(
+        thread: Thread,
+        originalException: Throwable
+    ) {
+        coroutineScope.launch(dispatcherProvider.io() + NonCancellable) {
+            try {
+                uncaughtExceptionRepository.recordUncaughtException(originalException, UncaughtExceptionSource.GLOBAL)
+                offlinePixelCountDataStore.applicationCrashCount += 1
+            } catch (e: Throwable) {
+                Timber.e(e, "Failed to record exception")
+            } finally {
+                originalHandler?.uncaughtException(thread, originalException)
+            }
+        }
+    }
```

Comment:
Should probably add a test for this, can copy the one from `AlertingUncaughtExceptionHandler`.

Reply:
Makes sense, Will add the tests